extensions_admin/pulp_docker/extensions/admin/upload.py

```diff
@@ -30,16 +31,33 @@ def __init__(self, context):
 
     def determine_type_id(self, filename, **kwargs):
         """
-        Determine whether it is a V1 or V2 image
+        Determine type id of the upload file by file type.
+
+        json -> Manifest List
+        tarfile -> Image or Manifest
 
         :return: ID of the type of file being uploaded
         :rtype:  str
+        :raises: RuntimeError if file is not a valid tarfile or json file.
         """
-        image_manifest = tarutils.get_image_manifest(filename)
-        if isinstance(image_manifest, list):
-            return constants.IMAGE_TYPE_ID
+        try:
+            with open(filename) as upload:
+                json.loads(upload.read())
+            return constants.MANIFEST_LIST_TYPE_ID
+        except ValueError:
+            pass
```

Comment:
hm, i am on the fence here, what about nest second try except under the ValueError?

Reply:
Since it is functionally equivalent, I'd like to leave it as is, since I find it easier to read this way.